pkg/broker/payments.go

```diff
@@ -39,15 +39,25 @@ func (p PaymentBroker) Run(started, stopped chan bool, stop chan context.Context
 					p.sendEvent(giga.BrokerEvent{Type: giga.NewInvoice, ID: e.ID})
 				case giga.InvoiceConfirmed:
 					// from Confirmer
-					err := p.store.Commit([]any{
-						giga.MarkInvoiceAsPaid{
-							InvoiceID: giga.Address(e.ID),
-						},
-					})
+					txn, err := p.store.Begin()
+					// XXX This loop needs bettre failure modes!
+					// this stuff needs to be buffered, perhaps WAL?
```

Comment:
Agree - but ChainFollower will take on this responsibility: it will only make forward progress if it can commit the effects to the db.